sdk/lib/pseh/i386/pseh3.c

```diff
@@ -184,23 +193,45 @@ _SEH3$_InvokeFilter(
 
 void
 __attribute__((regparm(1)))
-_SEH3$_AutoCleanup(
+_SEH3$_C_AutoCleanup(
     volatile SEH3$_REGISTRATION_FRAME *Frame)
 {
     _SEH3$_Unregister(Frame);
 
     /* Check for __finally frames */
     if (Frame->ScopeTable->Target == NULL)
     {
-       _SEH3$_InvokeFilter(Frame, Frame->ScopeTable->Filter);
+#ifdef __clang__
+        _SEH3$_InvokeFilter(Frame, Frame->ScopeTable->Filter, _SEH3$_CLANG_HANDLER);
+#else
+        _SEH3$_InvokeFilter(Frame, Frame->ScopeTable->Filter, _SEH3$_NESTED_HANDLER);
+#endif
     }
+}
 
+void
+__attribute__((regparm(1)))
+_SEH3$_CPP_AutoCleanup(
+    volatile SEH3$_REGISTRATION_FRAME *Frame)
+{
+    if (Frame->Handler)
+        _SEH3$_UnregisterFrame(Frame);
+    else
+        _SEH3$_UnregisterTryLevel(Frame);
```

Comment:
Are these 4 lines OK, or should they also be replaced by the more generic `_SEH3$_Unregister(Frame);` ? If not, why?